### PR TITLE
Emmett shooting

### PR DESCRIPTION
**Pull request by Emmett-Byrne**

### CODE REVIEW

File: ARGO/include/ProjectileManager.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+	Entity entity;
+	int type;
+};
+
+class ProjectileManager
+{
+public:
+	ProjectileManager();
+	void createPlayerBullet(createBulletEvent event);
```
Comment (CathalRedmond) [inline ARGO/include/ProjectileManager.h]: t_

File: ARGO/include/ProjectileManager.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	void createEnemyBullet(createBulletEvent event);
```
Comment (CathalRedmond) [inline ARGO/include/ProjectileManager.h]: t_

**Review by SeanAbnerNash — APPROVED** [review]

File: ARGO/src/ForceComponent.cpp [hunk 3]
```diff
@@ -2,15 +2,17 @@
 #include "ForceComponent.h"
 
 ForceComponent::ForceComponent() :
```
Comment (CathalRedmond) [inline ARGO/src/ForceComponent.cpp]: do you need this anymore. You can just set the values to have default values like you have for has friction in the overloaded constructor

File: ARGO/src/ProjectileManager.cpp [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ProjectileManager::ProjectileManager() :
+	m_nextEnemyBullet(0),
+	m_nextPlayerBullet(0)
+{
+	for (auto& bullet : m_playerBullets)
+	{
+		bullet.entity.addComponent(new TransformComponent());
+		bullet.entity.addComponent(new ColourComponent(glm::linearRand(0, 255), glm::linearRand(0, 255), glm::linearRand(0, 255), 255));
+		bullet.entity.addComponent(new ForceComponent(glm::vec2(0,0), false));
+		bullet.entity.addComponent(new HealthComponent(1, 0));
```
Comment (CathalRedmond) [inline ARGO/src/ProjectileManager.cpp:13]: magic numbers - cant tell what youre passing in

Comment (CathalRedmond) [reply ARGO/src/ProjectileManager.cpp:13]: same on line 22

File: ARGO/src/TimerComponent.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+bool TimerComponent::tick(float t_time)
+{
+	if (m_currentTime > 0)
+	{
+		m_currentTime -= t_time;
+		return true;
+	}
+	else return false;
```
Comment (CathalRedmond) [inline ARGO/src/TimerComponent.cpp]: dont need else here.

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

**Pull request merged**